projects/plugins/protect/src/js/components/threats-list/pagination.jsx

```diff
@@ -0,0 +1,159 @@
+import { Button } from '@automattic/jetpack-components';
+import React, { useCallback, useEffect, useState, useMemo, memo } from 'react';
+import styles from './styles.module.scss';
+
+const PaginationButton = memo( ( { pageNumber, currentPage, onPageChange } ) => {
+	const handleClick = useCallback( () => {
+		onPageChange( pageNumber );
+	}, [ onPageChange, pageNumber ] );
+
+	return (
+		<Button
+			onClick={ handleClick }
+			variant={ currentPage === pageNumber ? 'primary' : 'secondary' }
+			aria-current={ currentPage === pageNumber ? 'page' : undefined }
+		>
+			{ pageNumber }
+		</Button>
+	);
+} );
+
+const Pagination = ( { list, itemPerPage = 10, children } ) => {
+	const [ currentPage, setCurrentPage ] = useState( 1 );
+
+	const totalPages = useMemo( () => Math.ceil( list.length / itemPerPage ), [ list, itemPerPage ] );
+
+	const currentItems = useMemo( () => {
+		const indexOfLastItem = currentPage * itemPerPage;
+		const indexOfFirstItem = indexOfLastItem - itemPerPage;
+		return list.slice( indexOfFirstItem, indexOfLastItem );
+	}, [ currentPage, list, itemPerPage ] );
+
+	const onPageChange = useCallback( pageNumber => {
+		setCurrentPage( pageNumber );
+	}, [] );
+
+	const [ isSmall, setIsSmall ] = useState( window.matchMedia( '(max-width: 1220px)' ).matches );
```

Comment:
Could `useBreakpointMatch` be useful here?

Reply:
I started with `useBreakpointMatch` but found that we seem to have a unique requirement with how our UI shifts in various viewports that was not exactly covered by the options it provided (`isSm`, `isMd`,` isLg`).